fix: changes for xserver implementation

robertbolgeo opened the pull request:
# Description

added build files so that the project can be added to the xserver hosting platform.


## Related Issues

Link to any GitHub issues or client requests being addressed in this PR [use"#"](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#referencing-issues-and-pull-requests). If no issue exists, please update [pull-request-board](https://github.com/orgs/pride-land/projects/1).

- Issue/Request: [#230 ](https://github.com/pride-land/frontend-pride-land/issues/230)


## Type of change

Please delete options that are not relevant. (All options may be deleted for minor changes)

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update


# How Has This Been Tested?

Extensive testing through cURL tests to make sure all expected endpoints and site URLs are working. 
Tested https security by using whynopadlock.com.
Site functionality is continuing to be tested by just clicking and messing around.


- [x] cURL testing: In order to understand front and backend URL connectivity I used cURL testing (Postman would work too) from the command terminal. For anyone looking to duplicate these tests, just choose the endpoint or access point in question and type 'curl https://pridefarm.org/path/to/endpoint/' 
- [x] Feel free to also test the https qualifications of these endpoints by visiting whynopadlock.com and putting the endpoint into the free tester there.

## Screenshots (if applicable)
Including an image of the .htaccess file that is necessary for xserver to serve the static files for the front end.
<img width="1578" alt="Screenshot 2024-12-07 at 15 45 07" src="https://github.com/user-attachments/assets/b0c8e922-a2bf-4ad6-bc05-341216bfdca0">

## Checklist
- [x] I have ensured the code adheres to the coding standards of the project.
- [ ] If changes require documentation updates, I have included them.
- [x] I have performed a self-review of my code.
- [x] New members can easily understand this code, and I have added comments for clarity.
- [x] The PR is client-ready (all client-related feedback has been addressed).

## Additional Context or Notes
Most of the relevant changes didn't actually occur in our code base. Most changes that this PR addresses were done on the xServer server panel. If there are questions on how to use this in the future I think a separate PR request for the necessary documentation for those changes will be necessary.
- This was a pretty hacky fix. There are a lot of separate pieces that could become incompatible if any of the services were to update to stop working together. Some testing to ensure that the fcgi and htaccess files are executable and working correctly would be a good idea. After everything else is smooth I will come back for that.   :clinking_glasses:

robertbolgeo committed:
fix: added build files so that the project can be added to the xserver hosting platform more easily.

NabbeunNabi commented on lib/cjs/src/Admin/admin-api/admin-layout.d.ts:
```diff
@@ -0,0 +1,7 @@
+import { Image } from "../admin-interface/AdminGalleryTypes";
+export declare const deleteImage: (image: Image) => Promise<any>;
+export declare const getAllImages: () => Promise<any>;
+export declare const getCardRefs: () => Promise<any>;
+export declare const addHeroTagToImg: (image: Image) => Promise<any>;
+export declare const removeHeroTagFromImg: (image: Image) => Promise<any>;
+export declare const getHeroImages: () => Promise<any>;
```
@robertbolgeo Why are these returning `any`?

I am cool with any if the data is controlled by an outside api. But if you know what it returns you should look at response types and maybe make a file that shows them.

Can be addressed in a future ticket. But curious.

robertbolgeo replied:
I've added a git issue for this problem #236

NabbeunNabi commented on lib/cjs/src/Admin/admin-api/admin-blogs.d.ts:
```diff
@@ -0,0 +1 @@
+export declare const postBlogs: (formData: FormData) => Promise<any>;
```
♻️

NabbeunNabi commented on lib/cjs/src/Admin/admin-api/admin-control.d.ts:
```diff
@@ -0,0 +1 @@
+export declare const fetchAllStaff: () => Promise<[]>;
```
Does this return an empty promise array?

NabbeunNabi commented on lib/cjs/src/Admin/admin-api/admin-feedbacks.d.ts:
```diff
@@ -0,0 +1,4 @@
+import { FeedbackStatus } from "../admin-interface/AdminFeedbackType";
+export declare const getFeedbacks: () => Promise<any>;
```
♻️

NabbeunNabi commented on lib/cjs/src/Admin/admin-api/admin-volunteers.d.ts:
```diff
@@ -0,0 +1,3 @@
+import { MinPutVolunteer } from "../admin-interface/AdminVolunteerType";
+export declare const getAllVolunteers: () => Promise<any>;
```
♻️

NabbeunNabi commented on lib/cjs/src/Admin/admin-pages/SideBar.d.ts:
```diff
@@ -0,0 +1,5 @@
+interface SideBar {
+    elt: any;
```
What is `elt`? I cannot tell by the variable name or the type. Could be something I am not familiar with?

NabbeunNabi commented on lib/cjs/src/Admin/admin-utils/PrivateRoute.d.ts:
```diff
@@ -0,0 +1,7 @@
+declare const PrivateRoute: ({ children }: {
+    children: ReactNode;
+}, { ...rest }: {
+    [x: string]: any;
```
♻️

NabbeunNabi commented on lib/cjs/src/api/feedbacks.d.ts:
```diff
@@ -0,0 +1,3 @@
+import FeedbackType from "../interfaces/FeedbackType";
+export declare const fetchAllFeedbacks: () => Promise<FeedbackType[] | undefined>;
+export declare const postFeedback: (formData: FormData) => Promise<any>;
```
♻️

NabbeunNabi commented on lib/cjs/src/api/fetchBinaryImage.d.ts:
```diff
@@ -0,0 +1 @@
+export declare const fetchAllBinaryImage: () => Promise<any>;
```
♻️

NabbeunNabi commented on lib/cjs/src/api/upload-images.d.ts:
```diff
@@ -0,0 +1,2 @@
+export declare const postImage: (formData: FormData) => Promise<any>;
+export declare const fetchAllImages: () => Promise<any>;
```
♻️

NabbeunNabi commented on lib/cjs/src/components/blogComponents/blogFocus.d.ts:
```diff
@@ -0,0 +1,9 @@
+import BlogsTypes from "../../interfaces/BlogsType";
+import React, { Dispatch, SetStateAction } from "react";
+interface Props {
+    chosenBlog: BlogsTypes;
+    blogsProp: BlogsTypes[];
+    setChosenBlog: Dispatch<SetStateAction<BlogsTypes | null>>;
```
I love this variable naming and typing!!!

NabbeunNabi commented on lib/cjs/src/components/blogComponents/pagination.d.ts:
```diff
@@ -0,0 +1,7 @@
+interface Props {
+    blogsPerPage: number;
+    length: number;
+    handlePagination: any;
```
♻️

NabbeunNabi commented on lib/cjs/src/components/homeComponents/Section.d.ts:
```diff
@@ -0,0 +1 @@
+export {};
```
Leaving a comment as to why you are exporting an empty object might be helping above this

robertbolgeo replied:
This is caused by an empty component file being compiled at build. The component has been destroyed.

NabbeunNabi commented on lib/cjs/src/game/EventBus.d.ts:
```diff
@@ -0,0 +1,2 @@
+import { Events } from 'phaser';
+export declare const EventBus: Events.EventEmitter;
```
🚌

NabbeunNabi commented on src/Admin/admin-api/admin-layout.tsx:
```diff
@@ -1,59 +1,57 @@
+const endpoint = process.env.backend_url;
+const media_url = process.env.media_url;
+
+export const deleteImage = async (image: Image) => {
+ const request = await fetch(endpoint + `medias/${image.id}/`, {
+  method: "DELETE",
+  headers: {
+   "Content-Type": "application/json",
+  },
+ });
+ const result = await request.json();
+ return result;
+};
```
Should use a try catch block to catch any errors and `console.error` so developers can see what the problems are

robertbolgeo replied:
made an issue to fix this #237

NabbeunNabi commented on src/Admin/admin-api/admin-layout.tsx:
```diff
@@ -1,59 +1,57 @@
+ return result;
+};
+
+export const getAllImages = async () => {
+ const request = await fetch(endpoint + "medias/");
+ const result = await request.json();
+ return result;
+};
```
♻️

NabbeunNabi commented on src/Admin/admin-api/admin-layout.tsx:
```diff
@@ -1,59 +1,57 @@
+ return result;
+};
+
+export const getCardRefs = async () => {
+ const request = await fetch(endpoint + "cardrefs/");
+ const result = await request.json();
+ return result;
+};
```
♻️

NabbeunNabi commented on src/Admin/admin-api/admin-layout.tsx:
```diff
@@ -1,59 +1,57 @@
+ return result;
+};
+
+export const addHeroTagToImg = async (image: Image) => {
+ const request = await fetch(endpoint + `medias/${image.id}/`, {
+  method: "PATCH",
+  headers: {
+   "Content-Type": "application/json",
+  },
+  body: JSON.stringify({ set_as_hero: true }),
+ });
+ const result = await request.json();
+ return result;
+};
```
♻️

NabbeunNabi commented on src/Admin/admin-api/admin-layout.tsx:
```diff
@@ -1,59 +1,57 @@
+ return result;
+};
+
+export const removeHeroTagFromImg = async (image: Image) => {
```
♻️

NabbeunNabi commented on src/Admin/admin-api/admin-layout.tsx:
```diff
@@ -1,59 +1,57 @@
+ return result;
+};
+
+export const getHeroImages = async () => {
```
♻️

NabbeunNabi commented on src/App.tsx:
```diff
@@ -21,65 +21,76 @@ import MushroomGame from "./pages/MushroomGame";
-  }
-  return <>{children}</>; //allow access if authenticated
+ if (!user) {
+  //redirect to login if user not authenticated
```
💯 great comment

NabbeunNabi commented on src/App.tsx:
```diff
@@ -21,65 +21,76 @@ import MushroomGame from "./pages/MushroomGame";
-      LanguageContext = createContext("jp");
-    }
-  }, []);
+ // Get the language from local storage so it doesn't change on refresh
+ useEffect(() => {
+  const lang = sessionStorage.getItem("lang");
+  if (lang) {
+   setCurrentLang(lang);
+   LanguageContext = createContext(lang);
+  } else {
+   setCurrentLang("jp");
+   LanguageContext = createContext("jp");
+  }
+ }, []);
```
The comment says local storage but you are using session storage. Actually I think in your use case local storage would be great. Usually the user of the computer would want it to be the same always for the site. And not have to set always on tab visit. And it isn't private data. Just a locale key so it is safe to put in local storage.

robertbolgeo replied:
I changed it.

NabbeunNabi commented on src/pages/Home.tsx:
```diff
@@ -2,49 +2,45 @@ import Blogs from "../components/homeComponents/Blogs";
+   {images && <HeroCarousel images={images} />}
+   <Slogan></Slogan>
+   <MissionStatement />
+   {/* <section id="cards" className={"min-h-80 flex flex-col px-4"}>
```
If not used in future ticket you can remove this. Always can add later

NabbeunNabi replied:
I see it was fixed in a different PR

NabbeunNabi commented:
@robertbolgeo reviewed! Looks much prettier 😁. Just a few things to address. Also I would update the title of the PR. It just says fix at the moment. You can add any title standard you want but it could follow conventional commits.

- Also your PR description says documentation is required but you haven't checked that it is updated. Just clarifying as to if it is done or if documentation is needed

robertbolgeo committed:
chore: update empty array type

I changed empty array to be type any so that the linter will pick it up for better visibility to be fixed when addressing a previously opened issue.

robertbolgeo committed:
chore: change sessionStorage to localStorage for i18n

robertbolgeo committed:
chore: delete empty component

NabbeunNabi reviewed (approved):
Thanks for making those tickets and making the necessary changes

robertbolgeo committed:
chore: delete empty files

robertbolgeo committed:
chore: build project

robertbolgeo commented:
Added new issues for things I couldn't fix tonight, and fixed everything else. Thanks Will!
